[S-K-Y-Light] iP

S-K-Y-Light opened the pull request:
# Sora

> They Can Take Your World, They Can Take Your Heart, Cut You Loose From All You Know. But If It's Your Fate, Then Every Step Forward Will Always Be A Step Closer To Home – Sora ([Kingdom Hearts 3](https://www.kingdomhearts.com/3/us/home/))

Sora stresses your mind by reminding things you need to do. It's,
- text-based (Unfriendly to GUI Users)
- easy to learn (After One Month of Usage)
- ~~FAST~~ Negligibly Fast to use (Runs in O(N!) Runtime)

All you need to do is,
1. download it from [here](https://github.com/S-K-Y-Light/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing Tasks
- [ ] Graphical User Interface (coming soon)
- [ ] Automatically Doing your Tasks (coming never)

---

If you are a Java Programmer, you can use it to practice your Java too. Here's a main method that throws a SoraException:
```
public class Main {
    public static void main(String[] args) {
        throw new SoraException();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

S-K-Y-Light committed:
Rename Duke.java to Sora.java
Change Sora greeting & exit message

S-K-Y-Light committed:
Add Echo Feature using Scanner

S-K-Y-Light committed:
Add Horiztonal Lines

S-K-Y-Light committed:
Add Indentations

S-K-Y-Light committed:
Implement List Creation & Display

S-K-Y-Light committed:
Add Task.java

S-K-Y-Light committed:
Privatize parameters & Add setDone(bool)

S-K-Y-Light committed:
Update Task.java Constructor

S-K-Y-Light committed:
Integrate Task.java & Add Marking / Unmarking Tasks

S-K-Y-Light committed:
Add Sora Lines to "Mark" & "Unmark"

S-K-Y-Light committed:
Create Task Subclasses Todo, Deadline & Event

S-K-Y-Light committed:
Modify Deadline & Event toString() & Declare Task as abstract

S-K-Y-Light committed:
Implement Todo, Deadline & Event

S-K-Y-Light committed:
Move Printed Statement on Total Amount of Tasks to Task Creation Commands Only

S-K-Y-Light committed:
Implement Automated Text UI Testing for Level 0 - 4

S-K-Y-Light committed:
Add Exceptions (Empty Command, Invalid Command & IndexOutOfBoundsException)

S-K-Y-Light committed:
Create TaskList.java

S-K-Y-Light committed:
Implement Delete Feature

S-K-Y-Light committed:
Update Automated Test Cases

S-K-Y-Light committed:
Fix Indentation for Deleted Task

S-K-Y-Light committed:
Remove TaskList as deemed unncessary

S-K-Y-Light committed:
Update Task.java to improve readability

S-K-Y-Light committed:
Separate each "if-else" to switch cases. Write functions for each case to improve readability. Add IndexOutOfBoundsException & NumberFormatException.

S-K-Y-Light committed:
Update Automated Test Cases

S-K-Y-Light committed:
Fix NoSuchElementException by adding bool isLive to Sora.java

S-K-Y-Light committed:
Fix Coding Standard

S-K-Y-Light committed:
Update Task.java & Subclasses to getTaskDetails

S-K-Y-Light committed:
Change SoraException.java to extend Exception

S-K-Y-Light committed:
Add "Save" & "Load" Feature

S-K-Y-Light committed:
Merge branch 'branch-Level-7'

S-K-Y-Light committed:
Add Date/Time Feature. Sora understands Date/Time in dd/MM/yy HHmm format.

S-K-Y-Light committed:
Merge branch 'branch-Level-8'

S-K-Y-Light committed:
Update Automated Test Cases

S-K-Y-Light committed:
Refactor Sora class. Add Parser, Storage, TaskList & Ui classes

S-K-Y-Light committed:
Merge branch 'branch-A-MoreOOP'

S-K-Y-Light committed:
Divide classes into packages

S-K-Y-Light committed:
Merge branch 'branch-A-Packages'

S-K-Y-Light committed:
Merge branch 'add-gradle-support'

S-K-Y-Light committed:
Add ParserTest.java & TaskListTest.java

S-K-Y-Light committed:
Update Gradle to create JAR File

S-K-Y-Light committed:
Remove Unnecessary Import Statements in Sora.java

S-K-Y-Light committed:
Implement Find Feature

S-K-Y-Light committed:
Add more Tasks in savedtasks.txt

S-K-Y-Light committed:
Document "Find" Methods

S-K-Y-Light committed:
Documentation

S-K-Y-Light committed:
Adhere to Coding Standard

S-K-Y-Light committed:
Merge branch 'branch-A-JavaDoc'

S-K-Y-Light committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/sora/Storage.java

S-K-Y-Light committed:
Change public method to private

S-K-Y-Light committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/sora/Tasks/Task.java

Nicholaskoh1 commented on src/main/java/sora/Parser.java:
```diff
@@ -0,0 +1,47 @@
+     */
+    public static ArrayList<String> parse(String command) {
+        ArrayList<String> parsedResult = new ArrayList<>();
+        String[] parse_1 = command.split(" ", 2);
```
Perhaps this could be better named? Maybe something like parseTemp or parsedArr instead of using _

S-K-Y-Light replied:
Nice Catch. Will Rename.

Nicholaskoh1 commented on src/main/java/sora/Storage.java:
```diff
@@ -0,0 +1,131 @@
+     * @param taskList TaskList.
+     * @return String of all Task Details divided by " | ".
+     */
+    private String saveTaskString(TaskList taskList) {
```
Perhaps this could be better named? I initially interpreted this as 'saving task strings' and not 'getting the task strings to be saved'.

S-K-Y-Light replied:
Sounds solid. Will Rename.

Nicholaskoh1 commented on src/main/java/sora/Tasks/Deadline.java:
```diff
@@ -0,0 +1,55 @@
+ */
+public class Deadline extends Task {
+    /** Deadline of the deadline in String Format */
+    protected String by;
```
Perhaps this could be better named? 'by' seems a bit ambigious on what it stands for.

Nicholaskoh1 commented on src/main/java/sora/Tasks/Task.java:
```diff
@@ -0,0 +1,71 @@
+     * @param s User's Substring.
+     * @return Boolean value.
+     */
+    public boolean descriptionContainsString(String s) {
```
Perhaps could be better named to suit a boolean method?

YANYAN-w replied:
Perhaps the method name can be changed to sound more like a boolean method?

Nicholaskoh1 reviewed (commented):
Just some nitpicks on the names of some of the methods and variables.

S-K-Y-Light committed:
Update README.md

S-K-Y-Light committed:
Update README.md

Update Download Link

S-K-Y-Light committed:
Update README.md

Update Download link

S-K-Y-Light committed:
Rename variables parse_1 & parse_2

S-K-Y-Light committed:
Rename saveTaskString

S-K-Y-Light committed:
Remove Wildcard Import

S-K-Y-Light committed:
Merge branch 'master' of https://github.com/S-K-Y-Light/ip

YANYAN-w commented on src/main/java/sora/Parser.java:
```diff
@@ -0,0 +1,47 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * Parser parses the User's input.
+ * It deals with making sense of the user command.
+ * It identifies mainCommand and additional details in the User's input.
+ */
```
I like that your JavaDocs is very comprehensive!

YANYAN-w commented on src/main/java/sora/Storage.java:
```diff
@@ -0,0 +1,131 @@
+     *
+     * @param taskList TaskList.
+     * @return String of all Task Details divided by " | ".
+     */
```
I like that the overhead comments for public methods are comprehensive!

YANYAN-w commented on src/main/java/sora/Storage.java:
```diff
@@ -0,0 +1,131 @@
+        switch (parsedFileTaskDetails[0]) {
+            case "T":
+                if (parsedFileTaskDetails.length != 3) {
+                    throw new SoraException("\tSora is unable to read file from line " + lineNumber);
```
Perhaps it is better to follow the coding standard that there is no indentation for case clauses 🤔

S-K-Y-Light replied:
Good Catch. I wonder how I missed that 🤔 
Will Fix 👍

S-K-Y-Light committed:
Fix Switch Cases Indentation in Storage.java

S-K-Y-Light committed:
Fix Switch Cases Indentation for Sora.java

S-K-Y-Light committed:
Add Style Checker for A-Checkstyle

S-K-Y-Light committed:
Create GUI for Sora

S-K-Y-Light committed:
Add Greeting when Sora is launched

S-K-Y-Light committed:
Add Exiting Sora App when user types "bye"

S-K-Y-Light committed:
Change Images of User & Sora

S-K-Y-Light committed:
Anchor TextField, Send Button & Scroll Pane

S-K-Y-Light committed:
Set limt to Window Size

S-K-Y-Light committed:
Add CSS Styling

S-K-Y-Light committed:
Change ImageView from Rectangle to Circle

S-K-Y-Light committed:
Fix Ui Output by adding "\n"

S-K-Y-Light committed:
Shorten Exit Time

S-K-Y-Light committed:
Merge branch 'branch-Level-10'

S-K-Y-Light committed:
Change ScrollBar from Black to Light Grey

S-K-Y-Light committed:
Add Documentation to Sora, Ui & TaskList

S-K-Y-Light committed:
Update TaskList FindTask Method to display "No Tasks Found"

S-K-Y-Light committed:
Fix Bug where User cannot Enter Command Keyword Only

S-K-Y-Light committed:
Add Assertions to Methods in Storage, Ui & TaskList

There are several methods containing crucial assumptions that should hold during execution.

Adding assertions ensures that the code is safer.

Let's continue to add assertions to other methods making crucial assumptions. Doing so will make the code safer in the future.

S-K-Y-Light committed:
Improve Code Quality

There are several instances of poor variable names and lack of comments on the WHAT & WHY aspects of the code.

Following the Code Quality Convention would greatly improve the code quality and readability of the code.

As a step towards code quality, let's ensure that code does not violate any of the Code Quality Conventions, such as repeated code, bad variable names, lack of comments and wrong code style.

S-K-Y-Light committed:
Add Comments on instances where streams are present

S-K-Y-Light committed:
Merge pull request #1 from S-K-Y-Light/branch-A-Assertions

Add Assertions to Methods in Storage, Ui & TaskList

S-K-Y-Light committed:
Merge branch 'master' into branch-A-CodeQuality

S-K-Y-Light committed:
Merge pull request #2 from S-K-Y-Light/branch-A-CodeQuality

Improve Code Quality

S-K-Y-Light committed:
Merge branch 'master' into branch-A-Streams

S-K-Y-Light committed:
Merge pull request #3 from S-K-Y-Light/branch-A-Streams

Add Comments on instances where streams are present

S-K-Y-Light committed:
Implement "Help" Feature for C-Help

S-K-Y-Light committed:
Upload Screenshot of Final iP

S-K-Y-Light committed:
Refactor GUI to gui

S-K-Y-Light committed:
Merge branch 'master' of https://github.com/S-K-Y-Light/ip

S-K-Y-Light committed:
Update README.md

Provide User Guide

S-K-Y-Light committed:
Update README.md

Improve & Update Sora User Guide

Greater Clarity is Provided.

Formatting has also been fixed.

Users should understand Sora User Guide easier.

S-K-Y-Light committed:
Update Sora's message for "Find" Feature

S-K-Y-Light committed:
Merge branch 'master' of https://github.com/S-K-Y-Light/ip

S-K-Y-Light committed:
Reformat GUI to gui in build.gradle
